packages/devtools_app/lib/src/screens/vm_developer/vm_object_model.dart

```diff
@@ -0,0 +1,149 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../service/vm_service_wrapper.dart';
+import '../../shared/globals.dart';
+import '../debugger/debugger_model.dart';
+
+/// Wrapper class for storing Dart VM objects with their relevant VM
+/// information.
+abstract class VmObject {
+  VmObject({required this.ref});
+
+  final ObjRef ref;
+
+  String? get name;
+
+  VmServiceWrapper get _service => serviceManager.service!;
+
+  IsolateRef? _isolate;
+
+  Obj? _obj;
+
+  Future<void> fetchObject() async {
+    final isolateRef = serviceManager.isolateManager.selectedIsolate.value!;
+    _isolate = await _service.getIsolate(isolateRef.id!);
+    _obj = await _service.getObject(_isolate!.id!, ref.id!);
+  }
+
+  Future<void> initialize();
```

Comment:
if fetchObject should always be called as part of initialize, we can mark this method with @mustCallSuper instead of making it abstract

```
@mustCallSuper
Future<void> initialize() async {
  await fetchObject();
}
```

then overrides of this method don't have to all call fetchObject, which will let you eliminate several of the overrides below to `initialize`